D/D.sublime-syntax

```diff
@@ -8,339 +8,2057 @@ file_extensions:
+
+  alias-in:
+    - match: '\b(alias)\b(?=[^;]*\bthis\s*;)'
+      scope: keyword.control.alias.d
+      push:  [alias-this-after, identifier]
+    - match: '\b(alias)\b'
+      scope: keyword.control.alias.d
       push:
-        - meta_scope: comment.block.d
-        - match: \*/
-          scope: punctuation.definition.comment.d
-          pop: true
-    - match: /\+
-      scope: punctuation.definition.comment.d
+        - match: '{{attribute_lookahead}}'
+          set: [alias-old, type]
+        - match: '\b({{name}})\s*(?=\(|=)'
+          captures:
+            1: entity.name.type.d
+          set: alias-new
+        - match: (?=\S)
+          set: [alias-old, type]
+  alias-this-after:
+    - match: \b(this)\b
+      scope: keyword.control.alias.d
+      set: expect-end-of-line
+    - include: not-whitespace-illegal-pop
+  alias-old:
+    - match: '\b({{name}})\b'
+      scope: entity.name.type.d
+      set:
+        - match: ','
+          scope: punctuation.separator.sequence.d
+          set: alias-old
+        - include: end-of-line
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  alias-new:
+    - match: '(?=\()'
+      push: function-argument-definition-list
+    - match: '(?==)'
+      set: alias-new-assign
+    - include: not-whitespace-illegal
+  alias-new-assign:
+    - match: '='
+      scope: keyword.operator.assignment.d
+      set: [alias-new-after-type, value]
+    - include: not-whitespace-illegal
+  alias-new-after-type:
+    - include: end-of-line
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set:
+        - match: '\b{{name}}\b'
+          scope: entity.name.type.d
+          set: alias-new
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+
+  class-in:
+    - match: '\b(class)\b'
+      scope: storage.type.class.d keyword.declaration.class.d
       push:
-        - meta_scope: comment.block.nested.d
-        - match: \+/
-          scope: punctuation.definition.comment.d
+        - match: '\b({{name}})\b'
+          scope: entity.name.class.d
+          set:
+            - match: '(?=\()'
+              set: [class-after, function-argument-definition-list]
+            - match: '(?=\S)'
+              set: class-after
+        - include: not-whitespace-illegal
+  class-after:
+    - include: end-of-line
+    - match: ':'
+      scope: punctuation.separator.mapping.d
+      set: [optional-template-body-block, base-class-list]
+    - match: '(?=\S)'
+      set: optional-template-body-block
+
+  base-class-list:
+    - match: ''
+      set: [base-class-list-after, basic-type]
+  base-class-list-after:
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: base-class-list
+    - match: '(?={)'
+      pop: true
+    - include: not-whitespace-illegal
+
+  interface-in:
+    - match: '\b(interface)\b'
+      scope: storage.type.interface.d keyword.declaration.interface.d
```

Comment:
Like a class, this should have `meta.interface` until at least the opening braces, but ideally the closing brace.